Create nswpdc7and8.xml

SmarterThanMe opened the pull request:
NSW PDC for Y11&12

If this is a debate format submission, please provide some information about the circuit, league or tournaments where this format is used:

This is the format for the Y11 and 12 division of the Premier's Debating Challenge run by the NSW Department of Education's Arts Unit.



By submitting this pull request, I agree to license my contributions under the [MIT License](https://github.com/czlee/debatekeeper-formats/blob/main/LICENCE.md).

czlee commented on v1/formats/nswpdc7and8.xml:
```diff
@@ -0,0 +1,57 @@
+  <prep-time length="60:00"/>
+  <speech-types>
+    <speech-type ref="speech" length="7:00" first-period="normal">
+      <bell time="4:00" number="1" next-period="pdc.1bell"/>
+      <bell time="6:00" number="2" next-period="pdc.2bell"/>
+      <bell time="finish" number="60" next-period="pdc.overtime"/>
```
Unless there's a particular reason to use custom colours, I'd recommend just using the built-in period types, to be consistent with other formats.
```suggestion
      <bell time="4:00" number="1" next-period="warning"/>
      <bell time="6:00" number="2" next-period="overtime"/>
      <bell time="finish" number="60" next-period="overtime"/>
```
But this raises another discussion about the speech format, which I left in a separate comment on this pull request.

SmarterThanMe replied:
I'd prefer to leave this as is (for tactical reasons). If you update the "grace period," so that it's editable within the format, I'll come back in and fix it up. Is that OK?

czlee replied:
Sorry I'm slow to come back to this—there are two issues that I kinda mangled together here, sorry: (1) the colours and period descriptions used, and (2) when the bells are set to ring. (2) is all good as it stands now, following our discussion. But for (1) I'd like to be a bit more consistent with other formats.

I've been trying to phase out the use of period descriptions in the normal case, leaving them only when the period demands description to keep track of what's happening. Examples ([here are all of them](https://github.com/search?q=repo%3Aczlee%2Fdebatekeeper-formats+period-type+path%3A%2F%5Ev1%5C%2Fformats%5C%2F%2F&type=code)):
- In NZ Officers' Cup, the first minute of prep time is used to choose a moot, and the second minute to choose a side, so those get labelled "Choose moot" and "Choose side"
- In Canadian formats the Leader of the Opposition can choose to speak last and speak for a single ten-minute block, rather than having separate substantive and rebuttal (what Oceania calls "reply") speeches. So at 7 minutes, the description "Rebuttal time" appears to indicate that rebuttal speech rules now apply.

As for colours, where warning bells are analogous I'm trying to use the built-in period types to keep colours consistent between formats. In this case I think we could use the existing `warning` (orange) and `overtime` (red) period types, but add a new `pdc.1bell` type to be yellow like you have? So something like

```suggestion
      <bell time="4:00" number="1" next-period="pdc.1bell"/>
      <bell time="6:00" number="2" next-period="warning"/>
      <bell time="finish" number="60" next-period="overtime"/>
```

Then only this period type is necessary (note no description):
``` xml
   <period-type ref="pdc.1bell">
     <name>PDC 1 Bell</name>
     <default-bgcolor>#CCAA00</default-bgcolor>
   </period-type>
```

Would that be sensible? I'll do this for all files and put it in a new branch/PR so you can see and try out the files.

(The default colours are here: https://github.com/czlee/debatekeeper/blob/master/app/src/main/assets/periods.xml)

czlee commented:
Hey @SmarterThanMe, thanks for submitting this, and great work figuring out why the validations were failing! I had a look at the error logs and am just now realising they're a bit cryptic 🧐

Question about the format—do I [understand that](https://artsunit.nsw.edu.au/program/the-premiers-debating-challenge#How%20long%20do) (for years 7 and 8) speeches are meant to be 6 minutes long, but the "grace period" before continuous ringing is 1 minute? (in contrast to say, British Parliamentary, which has a 15-second grace period) Or are speakers just expected to speak until 7 minutes?

If it's the former, this might be reflect a suboptimal design with the app. The app has a concept of "grace period"—but it's user-configurable, not format-dependent. (Go to Settings, then set "First overtime bell" to 60 seconds and "Overtime bell period" to 3 seconds.) Perhaps this should be changed to being format-dependent.

But for now (and possibly for the foreseeable future) the way you've done this file is an adequate workaround. I just wanted to ask to aid my understanding. I guess the practical issue with this workaround is that the screen says "7 minutes long" at the bottom, because that's what it looks like in the file.

czlee commented on v1/formats/nswpdc7and8.xml:
```diff
@@ -0,0 +1,57 @@
+  <version>1</version>
+  <info>
+    <region>NSW, Australia</region>
+    <level>Primary School</level>
+    <used-at>NSW Premier's Debating Challenge (for Years 7 and 8)</used-at>
```
```suggestion
    <level>Years 7 and 8</level>
    <used-at>NSW Premier's Debating Challenge</used-at>
```
(these are just arbitrary strings that aren't used for anything other than display, but this seems more descriptive to me?)

czlee commented on v1/formats/nswpdc7and8.xml:
```diff
@@ -0,0 +1,57 @@
+    <region>NSW, Australia</region>
+    <level>Primary School</level>
+    <used-at>NSW Premier's Debating Challenge (for Years 7 and 8)</used-at>
+    <description>3 speakers + 4th vs 3 speakers + 4th, 6 minute speeches</description>
```
Being a bit picky but I'd use something similar to the Australian Easters description, like "3 vs 3, 6-minute speeches, no replies, no POIs". If you want to acknowledge the fourth prep teammate we could, but the World Schools format doesn't do so and the purpose is mostly to describe the timings/format rather than prep rules.

czlee reviewed (commented):
Thanks again! The file works fine as it is, I'm just picky with the conformity of files in the official repository 😊

SmarterThanMe commented:
*> Question about the format—do I understand that
<https://artsunit.nsw.edu.au/program/the-premiers-debating-challenge#How%20long%20do>
(for
years 7 and 8) speeches are meant to be 6 minutes long, but the "grace
period" before continuous ringing is 1 minute? (in contrast to say, British
Parliamentary, which has a 15-second grace period) Or are speakers just
expected to speak until 7 minutes?*

That is correct. Strictly speaking, for all of the divisions, the time
limit is the time limit. However, speakers are allowed to conclude after
the two bells (and, in fact, I coach them to do so - but some of my
colleagues think I push it 😅 - that's why I used the other colours ).

So, adjudicators will listen after the time limit, during the "grace
period" and then stop (and the rules require them to stop) listening at the
continuous bells.

I saw that you can set a grace period, but being that I train my teams to
go for the continuous bells, I thought it would be better to just set it up
differently.

- daniel judd

On Mon, 30 Oct 2023 at 22:11, Chuan-Zheng Lee ***@***.***>
wrote:

> Hey @SmarterThanMe <https://github.com/SmarterThanMe>, thanks for
> submitting this, and great work figuring out why the validations were
> failing! I had a look at the error logs and am just now realising they're a
> bit cryptic 🧐
>
> Question about the format—do I understand that
> <https://artsunit.nsw.edu.au/program/the-premiers-debating-challenge#How%20long%20do>
> (for years 7 and 8) speeches are meant to be 6 minutes long, but the "grace
> period" before continuous ringing is 1 minute? (in contrast to say, British
> Parliamentary, which has a 15-second grace period) Or are speakers just
> expected to speak until 7 minutes?
>
> If it's the former, this might be reflect a suboptimal design with the
> app. The app has a concept of "grace period"—but it's user-configurable,
> not format-dependent. (Go to Settings, then set "First overtime bell" to 60
> seconds and "Overtime bell period" to 3 seconds.) Perhaps this should be
> changed to being format-dependent.
>
> But for now (and possibly for the foreseeable future) the way you've done
> this file is an adequate workaround. I just wanted to ask to aid my
> understanding. I guess the practical issue with this workaround is that the
> screen says "7 minutes long" at the bottom, because that's what it looks
> like in the file.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/czlee/debatekeeper-formats/pull/30#issuecomment-1784965297>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEAJZRKRN2VOQW5SF72H45TYB6DNVAVCNFSM6AAAAAA6UQOZB2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOBUHE3DKMRZG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

SmarterThanMe commented on v1/formats/nswpdc7and8.xml:
```diff
@@ -0,0 +1,57 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<debate-format schema-version="2.2">
+  <name>NSW Premier's Debating Challenge (7 and 8)</name>
+  <short-name>NSWPDC7and8</short-name>
```
Agree. I'll make that change, and also to the other files.

SmarterThanMe commented:
I'll update the changes that you've suggested here on the other files as well (except that I'd greatly prefer to keep the timing as it is, if you're alright with that).

SmarterThanMe commented:
Legend. I think I've fixed the issues that you commented on in all of the
files I put up. I've added an issue for the format configurable grace
period (is this what I was meant to do?).

When I get 10 minutes, I'll sort out formats for the other NSW and
Australian competitions for which I know the rules.

- daniel judd

On Mon, 30 Oct 2023 at 22:11, Chuan-Zheng Lee ***@***.***>
wrote:

> Hey @SmarterThanMe <https://github.com/SmarterThanMe>, thanks for
> submitting this, and great work figuring out why the validations were
> failing! I had a look at the error logs and am just now realising they're a
> bit cryptic 🧐
>
> Question about the format—do I understand that
> <https://artsunit.nsw.edu.au/program/the-premiers-debating-challenge#How%20long%20do>
> (for years 7 and 8) speeches are meant to be 6 minutes long, but the "grace
> period" before continuous ringing is 1 minute? (in contrast to say, British
> Parliamentary, which has a 15-second grace period) Or are speakers just
> expected to speak until 7 minutes?
>
> If it's the former, this might be reflect a suboptimal design with the
> app. The app has a concept of "grace period"—but it's user-configurable,
> not format-dependent. (Go to Settings, then set "First overtime bell" to 60
> seconds and "Overtime bell period" to 3 seconds.) Perhaps this should be
> changed to being format-dependent.
>
> But for now (and possibly for the foreseeable future) the way you've done
> this file is an adequate workaround. I just wanted to ask to aid my
> understanding. I guess the practical issue with this workaround is that the
> screen says "7 minutes long" at the bottom, because that's what it looks
> like in the file.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/czlee/debatekeeper-formats/pull/30#issuecomment-1784965297>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEAJZRKRN2VOQW5SF72H45TYB6DNVAVCNFSM6AAAAAA6UQOZB2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOBUHE3DKMRZG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

SmarterThanMe commented:
G'day CZLee,

Are there any final hurdles for this to overcome for these formats (I'm
still keen to get the formats done for the other competitions too) to be
included in the list accessible through the app?

I'm doing a presentation to some of the school and regional level debating
coordinators in a few weeks here in Sydney, and I'd really like to give
your work a bit of a plug. :)

- daniel judd

On Mon, 30 Oct 2023 at 22:50, Daniel Judd ***@***.***> wrote:

> Legend. I think I've fixed the issues that you commented on in all of the
> files I put up. I've added an issue for the format configurable grace
> period (is this what I was meant to do?).
>
> When I get 10 minutes, I'll sort out formats for the other NSW and
> Australian competitions for which I know the rules.
>
> - daniel judd
>
> On Mon, 30 Oct 2023 at 22:11, Chuan-Zheng Lee ***@***.***>
> wrote:
>
>> Hey @SmarterThanMe <https://github.com/SmarterThanMe>, thanks for
>> submitting this, and great work figuring out why the validations were
>> failing! I had a look at the error logs and am just now realising they're a
>> bit cryptic 🧐
>>
>> Question about the format—do I understand that
>> <https://artsunit.nsw.edu.au/program/the-premiers-debating-challenge#How%20long%20do>
>> (for years 7 and 8) speeches are meant to be 6 minutes long, but the "grace
>> period" before continuous ringing is 1 minute? (in contrast to say, British
>> Parliamentary, which has a 15-second grace period) Or are speakers just
>> expected to speak until 7 minutes?
>>
>> If it's the former, this might be reflect a suboptimal design with the
>> app. The app has a concept of "grace period"—but it's user-configurable,
>> not format-dependent. (Go to Settings, then set "First overtime bell" to 60
>> seconds and "Overtime bell period" to 3 seconds.) Perhaps this should be
>> changed to being format-dependent.
>>
>> But for now (and possibly for the foreseeable future) the way you've done
>> this file is an adequate workaround. I just wanted to ask to aid my
>> understanding. I guess the practical issue with this workaround is that the
>> screen says "7 minutes long" at the bottom, because that's what it looks
>> like in the file.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/czlee/debatekeeper-formats/pull/30#issuecomment-1784965297>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AEAJZRKRN2VOQW5SF72H45TYB6DNVAVCNFSM6AAAAAA6UQOZB2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOBUHE3DKMRZG4>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

czlee commented:
@SmarterThanMe I meant to do this during the break and it kept getting pushed under other things sorry, will look at it next weekend (if not before)!

czlee commented on v1/formats/nswpdc7and8.xml:
```diff
@@ -0,0 +1,57 @@
+    <speech-type ref="speech" length="7:00" first-period="normal">
+      <bell time="4:00" number="1" next-period="pdc.1bell"/>
+      <bell time="6:00" number="2" next-period="pdc.2bell"/>
+      <bell time="finish" number="60" next-period="pdc.overtime"/>
```
One more thing... this makes me nervous, I don't think I had intended for this to be how this would work, and it seems to interfere with the "flash screen" bells (you can enable these in settings). I gather you want continuous bells to be standard across your competition, though?

If you're trying to enforce everyone having continuous bells, then I can tolerate this, just be aware that it'll do funky things. For example, I think the bell logic cancels all previous bells when a new bell starts, so if a user's own grace period is configured to (say) 15 seconds, in practice you'll get continuous bells until 15 seconds, and then the new overtime bell will take over, even though the 60 bells requested by the format hasn't finished. And if you have flash screen enabled... ngl, I don't actually know how it's handling this situation.

(I guess this bolsters the case that overtime bells ought to be format-specified, not user-configurable!)

czlee commented:
See #35 and #36 for combined versions of all four of #30, #31, #32, #33, with my suggested changes from comments in this PR worked in 🙂 

Thanks for your patience @SmarterThanMe!

czlee commented:
Included in #36.
